File: DailyRPG/Models/Contract.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DailyRpg.Models
{
    public class Contract
    {
        public  int Id { get; set; } = 0;
        public required string Title { get; set; }
        public string? Descricao { get; set; }
        public required int XpReward { get; set; } = 100;
        public required int CoinReward { get; set; } = 100;
        public DateTime StartDate { get; set; }

        public DateTime? CompletedAt { get; set; }
        public required bool IsCompleted { get; set; } = false;
        public string Difficulty { get; set; } = "medium";
        [Required]
        public int HunterUserId { get; set; }
        
        [ForeignKey("HunterUserId")]
        public virtual HunterUser HunterUser { get; set; } = null!;
    }
}

File: frontend/lib/models/contract.dart
class Contract {
  final String id;
  final String title;
  final String description;
  final int xpReward;
  final int coinReward;
  final bool isCompleted;
  final DateTime? startDate; 
  final String difficult;  

  Contract({
    required this.id,
    required this.title,
    required this.description,
    required this.xpReward,
    required this.coinReward,
    required this.isCompleted,
    required this.difficult, 
    this.startDate,         
  });

  factory Contract.fromJson(Map<String, dynamic> json) {
    return Contract(
      

      id: json['id']?.toString() ?? '', 
      
      title: json['title']?.toString() ?? 'Sem Título',
      
      description: json['descricao']?.toString() ?? 'Sem Descrição', 
      
      xpReward: json['xpReward'] ?? 0,
      coinReward: json['coinReward'] ?? 0,
      isCompleted: json['isCompleted'] ?? false,
      
      startDate: json['startDate'] != null 
          ? DateTime.parse(json['startDate']) 
          : null,

      difficult: json['difficult']?.toString() ?? 'Normal',
    );
  }
}